File: BaekJoon/Silver/1149/Python/1149.py
# no. 1149: RGB거리 (Silver I)
import sys

n = int(sys.stdin.readline())
memo = [[0 for _ in range(3)] for _ in range(n)]
memo[0] = [cost for cost in map(int, sys.stdin.readline().split())]

for i in range(1, n):
    # 현재가 r인경우, 이전에 g or b 을 선택한 경우의 수 중 최소값을 선택해야함
    cur_cost = [cost for cost in map(int, sys.stdin.readline().split())]
    memo[i][0] = cur_cost[0] + min(memo[i-1][1], memo[i-1][2])
    memo[i][1] = cur_cost[1] + min(memo[i-1][0], memo[i-1][2])
    memo[i][2] = cur_cost[2] + min(memo[i-1][0], memo[i-1][1])

print(min(memo[n-1]))

File: BaekJoon/Silver/1149/C++/1149.cpp
// no.1149: RGB 거리 (S1)

#include <cstdio>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    vector<vector<int>> cost(n, vector<int>(3, 0));
    for(int i=0; i<n; i++)
        for(int j=0; j<3; j++)
            scanf("%d", &cost[i][j]);
    vector<vector<int>> dp(n, vector<int>(3, 0));
    dp[0] = cost[0];
    for(int i=1; i<n; i++) {
        dp[i][0] = min(dp[i-1][1], dp[i-1][2])+cost[i][0];
        dp[i][1] = min(dp[i-1][0], dp[i-1][2])+cost[i][1];
        dp[i][2] = min(dp[i-1][0], dp[i-1][1])+cost[i][2];
    }
    printf("%d\n", min({dp[n-1][0], dp[n-1][1], dp[n-1][2]}));
    return 0;
}